ramips: use OKLI loader for TP-Link RE200 v1 and RE210 v1

DragonBluep opened the pull request:
Fixes: https://github.com/openwrt/openwrt/issues/16296

**bootlog:**
```
3: System Boot system code via Flash.
## Booting image at bc020000 ...
   Uncompressing Kernel Image ... OK
No initrd
## Transferring control to Linux (at address 80000000) ...
## Giving linux memsize in MB, 64

Starting kernel ...

OpenWrt kernel loader for MIPS based SoC
Copyright (C) 2011 Gabor Juhos <juhosg@openwrt.org>
Looking for OpenWrt image... found at 0xbc021000
Decompressing kernel... done!
Starting kernel at 80000000...

[    0.000000] Linux version 6.6.52 
```

musashino205 commented on target/linux/ramips/dts/mt7620a_tplink_re2x0-v1.dtsi:
```diff
@@ -63,9 +63,11 @@
 				label = "firmware";
 				reg = <0x20000 0x7c0000>;
+				openwrt,offset = <0x1000>;
+				openwrt,ih-magic = <IH_MAGIC_OKLI>;
```
CI reports the following error.

```
Error: ../dts/mt7620a_tplink_re2x0-v1.dtsi:70.25-26 syntax error
FATAL ERROR: Unable to parse input tree
```

`#include <dt-bindings/mtd/partitions/uimage.h>` needs to be added to use `IH_MAGIC_OKLI`.

DragonBluep replied:
Fixed. I restored the necessary change when I commit the final patch.

DragonBluep committed:
ramips: introduce TP-Link v1 header OKLI image recipe

It can be used to workaround the booting stuck issue caused by the
u-boot LZMA decompression error.

The new kernel image structure:
+------+------------------+------------------+---------------+-----------------+
| name | tplink-v1 header | OKLI lzma-loader | uImage header | lzma kernel+dtb |
+------+------------------+------------------+---------------+-----------------+
| size |      0x200       |       0xe00      |     0x40      |     dynamic     |
+------+------------------+------------------+---------------+-----------------+

Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Link: https://github.com/openwrt/openwrt/pull/16473
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
ramips: use OKLI loader for TP-Link RE200 v1 and RE210 v1

Using OKLI image to fix the booting stuck issue.

Tested with u-boot extracted from TP-Link stock images
"RE200(EU)_V1_171206.zip" and "RE210(US_CA)_V1_171205.zip".

Fixes: https://github.com/openwrt/openwrt/issues/16296
Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Link: https://github.com/openwrt/openwrt/pull/16473
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
